x/zoneconcierge/keeper/header_handler.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/babylonlabs-io/babylon/x/zoneconcierge/types"
+)
+
+// HandleHeaderWithValidCommit handles a CZ header with a valid QC
+func (k Keeper) HandleHeaderWithValidCommit(ctx context.Context, txHash []byte, header *types.HeaderInfo, isOnFork bool) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	babylonHeader := sdkCtx.HeaderInfo()
+	indexedHeader := types.IndexedHeader{
+		ConsumerId:          header.ClientId,
+		Hash:                header.AppHash,
+		Height:              header.Height,
+		Time:                &header.Time,
+		BabylonHeaderHash:   babylonHeader.AppHash,
+		BabylonHeaderHeight: uint64(babylonHeader.Height),
+		BabylonEpoch:        k.GetEpoch(ctx).EpochNumber,
+		BabylonTxHash:       txHash,
+	}
+
+	k.Logger(sdkCtx).Debug("found new IBC header", "header", indexedHeader)
+
+	var (
+		chainInfo *types.ChainInfo
+		err       error
+	)
+	if !k.HasChainInfo(ctx, indexedHeader.ConsumerId) {
+		// chain info does not exist yet, initialise chain info for this chain
+		chainInfo, err = k.InitChainInfo(ctx, indexedHeader.ConsumerId)
+		if err != nil {
+			panic(fmt.Errorf("failed to initialize chain info of %s: %w", indexedHeader.ConsumerId, err))
```

Comment:
all the panics here only can happens due to a programming error?

Reply:
Yes